### DISCUSSION TITLE
[Q&A]  多个 API Key 的输入页面，是否是功能重叠了？

### DISCUSSION THREAD

**Discussion by i-iooi-i** [lobehub/lobehub#302] [General Question] [answered]
### 🧐 问题描述 | Proposed Solution

![image](https://github.com/lobehub/lobe-chat/assets/1827701/f01406c3-47d1-436a-94a2-9dd21e2fb95c)
![image](https://github.com/lobehub/lobe-chat/assets/1827701/3ba648a8-dc04-4cc5-bbe5-2545224f09c7)

如题，现在存在两个key的输入页面，是否是功能重叠了，还是另有用途呢？
就在刚才把 lobe chat 给同事使用，他对这两处的输入位置，存在一些困惑。
若是功能重叠话的话，直接把这项功能放在 “设置” 页面，似乎更合理一些。

**Comment by arvinxx** [comment, marked as answer]
> 若是功能重叠话的话，直接把这项功能放在 “设置” 页面，似乎更合理一些。

> <img width="716" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/94f91856-2e82-4441-854f-7b1aa054ac3f">

去掉消息部分的设置的话，会导致大部分新人用户不知道怎么开始上手试用。这部分的目的是为了更好地帮助用户开始上手。

**Comment by i-iooi-i** [comment]
![image](https://github.com/lobehub/lobe-chat/assets/1827701/b9e971f7-4372-43ac-a104-783fbb7d42a0)
![image](https://github.com/lobehub/lobe-chat/assets/1827701/83777c39-491f-4c10-9f4b-cad723af0b82)
![image](https://github.com/lobehub/lobe-chat/assets/1827701/0b55cf1d-f2cd-42f9-a074-87d5944e0918)
![image](https://github.com/lobehub/lobe-chat/assets/1827701/aae1f1a7-441b-4782-9700-7ea0325d60a3)
![image](https://github.com/lobehub/lobe-chat/assets/1827701/13673d22-c382-4441-a101-4949ae99d3ca)
![image](https://github.com/lobehub/lobe-chat/assets/1827701/efdcf2ad-5d0c-43aa-859f-27b6443d8189)

找了几个项目的界面，都是大同小异。说一下我个人的感受哈，我是觉得用户这几年已经养成了使用习惯了，一般遇到这类chatgpt的界面，都会自觉的去界面左下角找设置图标，然后进去填写key的。

![image](https://github.com/lobehub/lobe-chat/assets/1827701/040a7143-e31a-4537-8b94-e606cd4a572d)
另外一方面，若是着眼国内用户的话，大家普遍使用的都是代理key。官方的这个填入进去，应该是会报错，无法使用的吧。

对于刚接触打开使用lobechat的用户来说，填写key，报错，是一种不好的体验，会有挫折感，可能就直接关闭页面走人了。
用户可能不会试着再摸索一下，看看左下角的设置里面，有没有填写代理的地方。

而且，重叠的两个功能，放在了两个入口，我觉得对用户理解上也确实会存在困扰。个人感受哈~

**Reply by arvinxx** [reply]
> 我是觉得用户这几年已经养成了使用习惯了

也就这半年。另外你的习惯基本上是因为持续在找类似工具，所以养成的习惯。但大部分普通用户，并不会有这种习惯的。

**Reply by arvinxx** [reply]
> 官方的这个填入进去，应该是会报错，无法使用的吧。

并不完全，其实若是试过 chat-preview.lobehub.com 这个，OpenAI 的 api key 填进去是正常可以 run 的

**Comment by leeyiding** [comment]
其实这样挺好的，这种场景一般都是自用或者给朋友分享使用，一般你在搭建时候就填好了apikey和密码，所以给朋友使用的话一般告诉个项目地址和密码就好了。如果说需要优化的地方，我觉得把密码放在第一个tab更合适，另外按你所说，可以在openai key配置界面再加一个代理地址的配置项，如果为空就保持默认

**Reply by arvinxx** [reply]
这个可以，我把 PROXY_URL 加上。

**Reply by arvinxx** [reply]
已在 https://github.com/lobehub/lobe-chat/commit/fa3170d7a1af55b3f7d03ef39264aeb2c2bffc4f 更新：

<img width="1043" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/79105758-e3c3-4b44-8f52-dac2da867827">

<img width="1030" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/60402c79-b635-48e4-b98b-874d881a572f">

**Comment by i-iooi-i** [comment]
是的，还有一个问题，是填写了密码，验证通过了。
然后又要弹窗出来让填写 key。
总感觉体验上不够简洁，操作上又有点冗余。 🙈

**Reply by arvinxx** [reply]
> 是的，还有一个问题，是填写了密码，验证通过了。
> 然后又要弹窗出来让填写 key。

如果是这样的话应该是 bug，什么情况下能复现？

**Reply by i-iooi-i** [reply]
![image](https://github.com/lobehub/lobe-chat/assets/1827701/ba831562-d60d-4690-bf3d-052a5ad5bcb1)
我是指这里哈，输入了密码，验证通过，然后就是右侧这个key停留在界面上，也要输入。

**Reply by canisminor1990** [reply]
@arvinxx 是不是把代理地址也放出来？

**Reply by i-iooi-i** [reply]
谢谢大佬，刚才拉取最新的镜像，测试了一下，该问题消失了哈，感谢！ ♥️♥️♥️
